catnip_uploader/catnip_uploader.py

```diff
@@ -1,36 +1,141 @@
+# Catnip Uploader is a Python script that allows you to upload firmware to CatSniffer boards V3.
+# This file is part of the CatSniffer project, http://electroniccats.com/
+# GNU GENERAL PUBLIC LICENSE
+# Version 3, 29 June 2007
+
+# Copyright (C) 2007 Free Software Foundation, Inc. https://fsf.org/
+# Everyone is permitted to copy and distribute verbatim copies
+# of this license document, but changing it is not allowed.
+
+
 import typer
 import serial
 import platform
 import requests
 import io
 import os
 import time
+import json
+import sys
 
 if platform.system() == "Windows":
     DEFAULT_COMPORT = "COM1"
+elif platform.system() == "Darwin":
+    DEFAULT_COMPORT = "/dev/tty.usbmodem0001"
 else:
     DEFAULT_COMPORT = "/dev/ttyACM0"
 
-GITHUB_URL_RELEASE       = "https://github.com/ElectronicCats/CatSniffer-Firmware/releases/download/board-v3.x-v1.0.0"
+GITHUB_RELEASE_URL               = "https://api.github.com/repos/JahazielLem/CatSniffer-Firmware/releases/latest"
```

Comment:
change for https://api.github.com/repos/ElectronicCats/CatSniffer-Firmware/releases/latest